ludwig/hyperopt/execution.py

```diff
@@ -737,27 +741,32 @@ def run_experiment_trial(config, local_hyperopt_dict, checkpoint_dir=None):
         run_experiment_trial_params = tune.with_parameters(run_experiment_trial, local_hyperopt_dict=hyperopt_dict)
         register_trainable(f"trainable_func_f{hash_dict(config).decode('ascii')}", run_experiment_trial_params)
 
-        analysis = tune.run(
-            f"trainable_func_f{hash_dict(config).decode('ascii')}",
-            config={
-                **self.search_space,
-                **tune_config,
-            },
-            scheduler=self.scheduler,
-            search_alg=search_alg,
-            num_samples=self.num_samples,
-            keep_checkpoints_num=1,
-            max_failures=1,  # retry a trial failure once
-            resources_per_trial=resources_per_trial,
-            time_budget_s=self.time_budget_s,
-            sync_config=self.sync_config,
-            local_dir=output_directory,
-            metric=metric,
-            mode=mode,
-            trial_name_creator=lambda trial: f"trial_{trial.trial_id}",
-            trial_dirname_creator=lambda trial: f"trial_{trial.trial_id}",
-            callbacks=tune_callbacks,
-        )
+        try:
+            analysis = tune.run(
+                f"trainable_func_f{hash_dict(config).decode('ascii')}",
+                config={
+                    **self.search_space,
+                    **tune_config,
+                },
+                scheduler=self.scheduler,
+                search_alg=search_alg,
+                num_samples=self.num_samples,
+                keep_checkpoints_num=1,
+                max_failures=1,  # retry a trial failure once
+                resources_per_trial=resources_per_trial,
+                time_budget_s=self.time_budget_s,
+                sync_config=self.sync_config,
+                local_dir=output_directory,
+                metric=metric,
+                mode=mode,
+                trial_name_creator=lambda trial: f"trial_{trial.trial_id}",
+                trial_dirname_creator=lambda trial: f"trial_{trial.trial_id}",
+                callbacks=tune_callbacks,
+            )
+        except Exception as e:
+            # Explicitly raise a ValueError if an error is encountered during a Ray trial. Otherwise, unit tests
+            # with failed trials silently hang forever.
+            raise ValueError(f"Encountered Ray Tune error: {e}")
```

Comment:
Not sure why we're raising a `ValueError` here, which is meant to denote that the user provided an invalid input. A `RuntimeError` would probably be more appropriate.

Also, the way we're re-raising the exception here is dropping the stack from the previous exception, not sure if that's intended. In general, we should use exception chaining here: https://peps.python.org/pep-3134/#explicit-exception-chaining

So something like:

```
raise RuntimeError("Encountered Ray Tune error") from e
```